src/components/NotesMindMap/NotesMindMapSelectors.js

```diff
@@ -0,0 +1,17 @@
+import { createSelector } from 'reselect';
+
+const getMindMapState = state => state.notesMindMap;
+
+export const getSelectedNote = createSelector(
+  getMindMapState,
+  mindMapState => mindMapState.selectedNote,
+);
+
+export const getSelectedNoteId = createSelector(getSelectedNote, selectedNote =>
```

Comment:
Why do we have part of selectors in `NotesMindMapContainer` and part in `NotesMindMapSelectors`?

Reply:
There is concept of layers of selectors:

1. reducer lever selector - implements some logic on reducer level
2. global level selector - includes all reducer level selectors and can be supplemented with some reusable selectors which combines reducer level selectors
3. component level selector - uses global selectors to select props for concrete component

It allows you to better organize your selectors, avoid circular dependencies (when your selectors use each other) and provides the way to reuse different modules selectors without applying some more complex technologies such as dependency injection.

In this example:

`NotesMindMapSelectors` - contains reducer level selectors
`NotesMindMapContainer` - contains component level selectors

Reply:
More detailed article:

https://cmichel.io/redux-selectors-structure